bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/BaseThingHandlerFactory.java

```diff
@@ -67,17 +66,13 @@ protected void activate(ComponentContext componentContext) {
     }
 
     /**
-     * Disposes the {@link BaseThingHandlerFactory}. If this method is
-     * overridden by a sub class, the implementing method must call <code>super.deactivate(componentContext)</code>
-     * first.
+     * Disposes the {@link BaseThingHandlerFactory}. If this method is overridden by a sub class, the implementing
+     * method must call <code>super.deactivate(componentContext)</code> first.
      *
      * @param componentContext
      *            component context (must not be null)
      */
     protected void deactivate(ComponentContext componentContext) {
-        for (ServiceRegistration<ThingHandler> serviceRegistration : this.thingHandlers.values()) {
```

Comment:
> This makes the implicit assumption that a ThingHandlerFactory does not have any dynamic dependencies, but that it is deactivated on any change of its dependent services. 
Correct - if service references are handed over to the handlers (what will be the case in general). Services only used by the factory could be dynamic.

> I think this must be clearly mentioned in the documentation
Yes, I need to make it obvious.